### DISCUSSION TITLE
How to finetune with a text file?

### DISCUSSION THREAD

**Discussion by ghost** [aub-mind/arabert#124] [Q&A] [answered]
Hey! Thanks for your amazing work on Arabic GPT-2.

I was wondering: is there a way to finetune the pretrained model using a text file?

I know Max Woolf already has this feature in [his Colab notebook](https://colab.research.google.com/drive/1VLG8e7YSEwypxU-noRNhsv5dW4NfTGce).
![image](https://user-images.githubusercontent.com/52135169/126936816-30679191-a3f0-4cca-8d6e-27a2166dd879.png)

Thank you!

**Comment by WissamAntoun** [comment, marked as answer]
Hey,

Can I suggest you try the pytorch notebook from huggingface here https://github.com/huggingface/notebooks/blob/master/examples/language_modeling.ipynb 

or you can check the single command training from the examples folder https://github.com/huggingface/transformers/tree/master/examples/pytorch/language-modeling

To run on your own training and validation files, use the following command:
```bash
python run_clm.py \
    --model_name_or_path gpt2 \
    --train_file path_to_train_file \
    --validation_file path_to_validation_file \
    --do_train \
    --do_eval \
    --output_dir /tmp/test-clm
```
This uses the built in HuggingFace Trainer for training. If you want to use a custom training loop, you can utilize or adapt the run_clm_no_trainer.py script. Take a look at the script for a list of supported arguments. An example is shown below:
```bash
python run_clm_no_trainer.py \
    --dataset_name wikitext \
    --dataset_config_name wikitext-2-raw-v1 \
    --model_name_or_path gpt2 \
    --output_dir /tmp/test-clm
```
